Cabal/src/Distribution/Simple/SrcDist.hs

```diff
@@ -504,10 +504,10 @@ printPackageProblems verbosity pkg_descr = do
       (errors, warnings) = partition isDistError (pureChecks ++ ioChecks)
   unless (null errors) $
       notice verbosity $ "Distribution quality errors:\n"
-                      ++ unlines (map explanation errors)
+                      ++ unlines (map show errors)
```

Comment:
Isn't introducing `show` here backwards?  The aim should be to reserve `show` for _Haskell_ representations of data, and use `pretty` or the like for user-friendly display.

Reply:
Indeed indeed! I would have removed that Show instance *and* `explanation` accessor which is at risk of becoming partial in the future.

I did not do it because the latter that would break `hackage-server` (they use `explanation`) and the former (maybe) other packages depending on Cabal (hackage-server? stack? some IDEs? — I cannot be sure about that because hunting `show`s in the wild is not as easy).

If we collectively decide it is a price worth paying, I will gladly do it! Indeed a `Show` that cannot be `Read` back is *not* haskelly at all!